frontend/packages/console-shared/src/components/badges/badge-factory.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import DevPreviewBadge from './DevPreviewBadge';
+import TechPreviewBadge from './TechPreviewBadge';
+
+export enum BadgeTypes {
+  DEV = 'DevPreview',
+  TECH = 'TechPreview',
+}
+
+export const getBadgeFromType = (badge: BadgeTypes): React.ReactElement => {
```

Comment:
@andrewballantyne In dev console pipeline pages we still use the badge componet directly. Since you have implemented the badge factory here, can we use this there also for consistency?

Reply:
It's an interesting point - we could go from:

```
badge={<DevPreviewBadge />}
```
to
```
badge={getBadgeFromType(PipelineRunModel.badge)}
```
In `frontend/packages/dev-console/src/components/pipelineruns/PipelineRunDetailsPage.tsx`

Reply:
When the model is updated, it'll address all places, not just generic pages.

Reply:
I didn't touch the `TechPreview` badges as I don't see models for all of them. But all the pipeline badges are tied together now.